src/credentials/FabricTable.cpp

```diff
@@ -269,6 +269,15 @@ CHIP_ERROR FabricInfo::SetCert(MutableByteSpan & dstCert, const ByteSpan & srcCe
     VerifyOrReturnError(srcCert.size() <= kMaxCHIPCertLength, CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrReturnError(CanCastTo<uint16_t>(srcCert.size()), CHIP_ERROR_INVALID_ARGUMENT);
 
+    uint32_t notValidBefore = 0;
+    ReturnErrorOnFailure(ExtractNotValidBeforeFromChipCert(srcCert, notValidBefore));
+
+    uint32_t notValidBeforeUTC;
+    VerifyOrReturnError(ChipEpochToUnixEpochTime(notValidBefore, notValidBeforeUTC), CHIP_ERROR_INVALID_TIME);
+
+    System::Clock::Seconds32 lkgt = System::Clock::Seconds32(notValidBeforeUTC);
+    System::SystemClock().SetClock_LastKnownGoodTime(std::chrono::duration_cast<System::Clock::Microseconds64>(lkgt));
```

Comment:
This has side-effects that are not trivial and may break future changes that try to optimize cert storage/loading. For instance, if we had proper "load certs only during CASE or access to NOC cluster" (which is the only requirement), then we would be in a situation where changing logic would lose last-known-good time.

A comment should be put here that last known good time setting should be retained.

Overall, this also has the side-effect of having a top-level policy that forces last known good time to only be based on an NOC, and to not follow the *maximum* of all certs.

I would recommend doing a separate strategy where we have a persistent last-known-good-time-from-certs that gets updated by *AddNOC* and *UpdateNOC*, which would always be a cached version of the "latest cert time".

This is still problematic if there is rotation of certs from controllers, where the only time available may be the time of the local certs, which will pre-date rotated certs.

Overall, last-known-good time is one thing to keep in mind, but the easiest solution would be to disregard time validity checks of certs in CASE if no time is available, unless we have much more methods of setting last-known-good time.